CHANGELOG.md

```diff
@@ -0,0 +1,107 @@
+# Changelog
+
+All notable changes to this project will be documented in this file.
+
+The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
+and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
+
+## [Unreleased]
+### Added
+ - API: New function to decode the image using a callback outputting a part of a
+   row per call.
+ - API: 16-bit float output support.
+
+### Changed
+ - Full JPEG XL part 1 specification support: Implemented all the spec required
+   to decode files to pixels, including cases that are not used by the encoder
+   yet. Part 2 of the spec (container format) is final but not fully implemented
+   here.
+ - Butteraugli metric improvements. Exact numbers are different from previous
+   versions.
+ - Memory reductions during decoding.
+ - Reduce the size of the jxl_dec library by removing dependencies.
+ - A few encoding speedups.
```

Comment:
Also I guess some (minor) decode speedups?

Could also mention the new speeds `thunder` and `lightning`, and the deprecation of `cjxl --speed` in favor of `cjxl --effort` (where it makes more sense that 3 is fast and 9 is slow).

Reply:
I'll add this in a follow up. We should have done this changelog.md before; and so far all of this goes to v0.4.x which I need to cherrypick; I'll work on adding the new stuff in main but not in v0.4.x (like the rewind api) on a follow up.